Azure/azure-storage-azcopy
# Resolve 'InvalidBlockList'

- Maintains the existing behavior by default
- Adds hidden flag '--delete-destination-file'.  When customers specify this flag, the existing destination blob is checked for uncommitted blocks and then is deleted. Then, the upload proceeds as normal.
- This works for Blob to Blob and File to Blob

Manual testing results:
1. Blob to Blob with flag set to **false** = Failed (400 The specified blob or block content is invalid.), original behavior 
2. Blob to Blob with flag set to **true** = Completed
3. File to Blob with flag set to **false** = Failed (400 The specified blob or block content is invalid.), original behavior 
4. File to Blob with flag set to **true** = Completed
5. Container with multiple Blobs (2 with uncommitted blocks + 1 normal block) with flag set to **true** = Completed 
6. Container with multiple Blobs (2 with uncommitted blocks + 1 normal block) with flag set to **false** = Failed (400 The specified blob or block content is invalid.), original behavior 

Added pos & neg unit test that tests DeleteDstBlob to ensure that the function works as intended.
